core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcherArguments.scala

```diff
@@ -95,6 +100,9 @@ private[mesos] class MesosClusterDispatcherArguments(args: Array[String], conf:
         "  -h HOST, --host HOST    Hostname to listen on\n" +
         "  -p PORT, --port PORT    Port to listen on (default: 7077)\n" +
         "  --webui-port WEBUI_PORT WebUI Port to listen on (default: 8081)\n" +
+        "  --webui-url  WEBUI_URL  The WebUI URL to be set in the underlying\n" +
```

Comment:
The code here is a bit strange it forces you to pass the master string for example with --master flag although you can also define the master url in a file (as spark.master) with the --properties-file option. 

Check here main args are parsed first and then any properties coming from a  file: https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcherArguments.scala#L33-L35

In the latter case we dont really need the --master flag yet i will fail if you dont provide it. The restriction is not there for --zk and spark.mesos.deploy.zookeeper.url.

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala#L98-L103